src/components/ToDo.tsx

```diff
@@ -0,0 +1,185 @@
+      })
+      .catch(() => dispatch({ type: ACTIONS.SET_ERROR, payload: 'Unable to delete a todo' }))
+  }
+
+  useEffect(() => {
+    if (editingTodo.current) {
+      editingTodo.current.focus();
+    }
+  });
+
+  function handleDoubleClickEdit(e: React.MouseEvent) {
+    e.preventDefault();
+    setIsEditing(true);
+    setEditingToDoData(todo.title);
+  }
+
+  function handleClick() {
+    setIsLoading(true);
+    updateTodo({
+      id: todo.id,
+      completed: !todo.completed,
+      title: todo.title,
+      userId: todo.userId,
+    })
+      .then(() => {
+        refreshLIst();
+      })
+      .catch(() => dispatch({ type: ACTIONS.SET_ERROR, payload: 'Unable to toggle a todo' }))
+  }
+
+  function sendUpdateRequest() {
+    updateTodo({
+      id: todo.id,
+      completed: todo.completed,
+      userId: todo.userId,
+      title: editingtoDoData,
+    })
+      .then(() => refreshLIst())
+      .catch(() => dispatch({ type: ACTIONS.SET_ERROR, payload: 'Unable to update a todo' }))
+  }
+
+  let result = false;
+
+  useMemo(() => {
+     result = true;
+    return result;
+  }, [editingtoDoData]);
+
+  function handleChanges() {
+    if (result){
+      if (editingtoDoData === '') {
+        deleteItem(todo.id);
+      }
+      else {
+        setIsLoading(true);
+        sendUpdateRequest();
+        setIsEditing(false);
+      }
+    }
+    setIsEditing(false);
+  }
+
+  function handleEnter(e: React.KeyboardEvent<HTMLInputElement>) {
+    if (e.nativeEvent.code === 'Enter' && e.target.value === '') {
+      deleteItem(todo.id);
+      setIsEditing(false);
+    } else if (e.nativeEvent.code === 'Enter') {
+      handleChanges();
+    }
+  }
+
+  return (
+    <div
+      className={classNames('todo', {
+      'completed': todo.completed,
+      editing: isEditing,
+    })}>
+      <label className="todo__status-label">
+        <input
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={handleClick}
+        />
+      </label>
+
+      {isLoading && (
+
+        <div className="modal overlay is-active">
+          <div className="modal-background has-background-white-ter" />
+          <div className="loader" />
+        </div>
+      )}
+      { (state.isLoading && todo.completed) && (
+        <div className="modal overlay is-active">
+        <div className="modal-background has-background-white-ter" />
+        <div className="loader" />
+      </div>
+      )}
+      { (state.toggleAll === 'completed' && todo.completed) && (
+        <div className="modal overlay is-active">
+        <div className="modal-background has-background-white-ter" />
+        <div className="loader" />
+      </div>
+      )}
+      { (state.toggleAll === 'active' && !todo.completed) && (
+        <div className="modal overlay is-active">
+        <div className="modal-background has-background-white-ter" />
+        <div className="loader" />
+      </div>
```

Comment:
we can combine all condition in one var and use smth like that
```suggestion
      {isLoading && (
        <div className="modal overlay is-active">
          <div className="modal-background has-background-white-ter" />
          <div className="loader" />
        </div>
      }
```

Reply:
done